pyvisfile/xdmf/__init__.py

```diff
@@ -0,0 +1,1169 @@
+            parent: Optional[Element] = None,
+            ):
+        """
+        :param parent: if provided, *self* is appended to the element.
+        """
+
+        super().__init__(parent, "Geometry", {
+            "Name": name,
+            "GeometryType": gtype.name,
+            })
+
+# }}}
+
+
+# {{{ time
+
+class TimeType(enum.Enum):
+    """Temporal information of the current grid."""
+    Single = enum.auto()
+    HyperSlab = enum.auto()
+    List = enum.auto()
+    Range = enum.auto()
+
+
+class Time(XdmfElement):
+    """
+    .. automethod:: __init__
+    """
+
+    def __init__(
+            self, *,
+            value: str,
+            parent: Optional[Element] = None,
+            ):
+        """
+        :param parent: if provided, *self* is appended to the element.
+        """
+
+        super().__init__(parent, "Time", {
+            "Value": value,
+            })
+
+# }}}
+
+
+# {{{ domain
+
+class Domain(XdmfElement):
+    """
+    .. automethod:: __init__
+    """
+
+    def __init__(
+            self, *,
+            name: Optional[str] = None,
+            parent: Optional[Element] = None,
+            ):
+        """
+        :param parent: if provided, *self* is appended to the element.
+        """
+
+        super().__init__(parent, "Domain", {
+            "Name": name,
+            })
+
+# }}}
+
+
+# {{{ information
+
+class Information(XdmfElement):
+    """
+    .. automethod:: __init__
+    """
+
+    def __init__(
+            self, *,
+            name: str,
+            value: str,
+            parent: Optional[Element] = None,
+            ):
+        """
+        :param parent: if provided, *self* is appended to the element.
+        """
+
+        super().__init__(parent, "Information", {
+            "Name": name,
+            "Value": value,
+            })
+
+# }}}
+
+
+# {{{ include
+
+class XInclude(XdmfElement):
+    """
+    .. automethod:: __init__
+    """
+
+    def __init__(
+            self, *,
+            href: Optional[str],
+            xpointer: Optional[str] = None,
+            parent: Optional[Element] = None,
+            ):
+        """
+        :param parent: if provided, *self* is appended to the element.
+        :param xpointer: path inside the file represented by *href*.
+        """
+        if xpointer is not None:
+            xpointer = f"xpointer({xpointer})"
+
+        super().__init__(parent, "xi:include", {
+            "href": href,
+            "xpointer": xpointer,
+            })
+
+# }}}
+
+# }}}
+
+
+# {{{ data arrays
+
+def _ndarray_to_string(ary):
+    if not isinstance(ary, np.ndarray):
+        raise TypeError(f"expected an 'ndarray', got '{type(ary).__name__}'")
+
+    ntype = _numpy_to_xdmf_number_type(ary.dtype)
+    if ntype == DataItemNumberType.Int or ntype == DataItemNumberType.UInt:
+        fmt = "%d"
+    elif ntype == DataItemNumberType.Float:
+        fmt = "%.16e" if ary.dtype.itemsize == 8 else "%.8e"
```

Comment:
Make this check each value, with an else for unsupported ones. (I'll wait for the first person trying to vis a `float128`. or a `float16`. Numpy can handle both.)

Reply:
Done in b8a6b07.